quickwit/quickwit-query/src/query_ast/term_query.rs

```diff
@@ -42,26 +43,43 @@ impl From<TermQuery> for QueryAst {
 }
 
 impl TermQuery {
-    #[cfg(test)]
-    pub fn from_field_value(field: impl ToString, value: impl ToString) -> Self {
-        Self {
-            field: field.to_string(),
-            value: value.to_string(),
+    fn get_tokenizer<'a>(&self, schema: &'a TantivySchema) -> Option<&'a str> {
+        let field = schema
+            .find_field(&self.field)
+            .or_else(|| schema.find_field(DYNAMIC_FIELD_NAME))?
+            .0;
+        match schema.get_field_entry(field).field_type() {
+            FieldType::Str(text_options) => Some(text_options.get_indexing_options()?.tokenizer()),
+            FieldType::JsonObject(json_options) => {
+                Some(json_options.get_text_indexing_options()?.tokenizer())
+            }
+            _ => None,
         }
     }
-}
 
-impl BuildTantivyAst for TermQuery {
-    fn build_tantivy_ast_impl(
+    pub(crate) fn ast_for_term_extraction(
         &self,
         schema: &TantivySchema,
         tokenizer_manager: &TokenizerManager,
-        _search_fields: &[String],
-        _with_validation: bool,
+    ) -> Result<TantivyQueryAst, InvalidQuery> {
+        self.build_ast(
+            schema,
+            tokenizer_manager,
+            self.get_tokenizer(schema)
+                .or(Some("raw"))
+                .map(ToString::to_string),
+        )
+    }
+
+    fn build_ast(
+        &self,
+        schema: &TantivySchema,
+        tokenizer_manager: &TokenizerManager,
+        tokenizer: Option<String>,
     ) -> Result<TantivyQueryAst, InvalidQuery> {
         let full_text_params = FullTextParams {
```

Comment:
I don't think this works.

What happens if your tokenizer generates two tokens?
You will end up making it into a disjunction.

so `roller-coaster` will match `roller-blade`... I'm not sure what the right spec is but this one is too naive.

Reply:
we could use `get_normalizer()` instead, which would just lowercase (if required), and never split tokens. That would make  `roller-coaster` match nothing instead of things it shouldn't. Properly matching would require not creating a TermSetQuery, but a bool query of what works in a termset on one side, and what doesn't on the other, and i'm not sure that's worth the extra trouble